feat!: switch to latest flat eslint config

janniks opened the pull request:
- Migrating eslint config to the new modern flat config files.
- Switch to flat configs
- Light repo maintenance 
- Update deps
- Switch to TS

> [!IMPORTANT]
> Probably important to squash this.

janniks committed:
fix!: switch to latest eslint

janniks committed:
ci: prerelease on develop

janniks committed:
build: remove yarn

janniks committed:
ci: update action versions

blockstack-devops commented:
:tada: This PR is included in version 3.0.0-develop.1 :tada:

The release is available on:
- [npm package (@develop dist-tag)](https://www.npmjs.com/package/@stacks/eslint-config/v/3.0.0-develop.1)
- [GitHub release](https://github.com/hirosystems/eslint-config/releases/tag/v3.0.0-develop.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

semantic-release-bot committed:
chore(release): 3.0.0-develop.1

## [3.0.0-develop.1](https://github.com/hirosystems/eslint-config/compare/v2.0.0...v3.0.0-develop.1) (2024-11-11)

### ⚠ BREAKING CHANGES

* switch to latest eslint

### Bug Fixes

* switch to latest eslint ([ab09dd9](https://github.com/hirosystems/eslint-config/commit/ab09dd92acf3f8be5782c2d078ab2f3ad8fac232))

janniks committed:
fix: add missing deps

janniks committed:
feat: pls prerelease semantic uwu

janniks committed:
docs: add readme

janniks committed:
ci: update workflow

semantic-release-bot committed:
chore(release): 3.0.0-develop.2

## [3.0.0-develop.2](https://github.com/hirosystems/eslint-config/compare/v3.0.0-develop.1...v3.0.0-develop.2) (2025-04-17)

### Features

* pls prerelease semantic uwu ([f71dc2c](https://github.com/hirosystems/eslint-config/commit/f71dc2c715bea6a2342dafaf1be7b51559aebac1))

### Bug Fixes

* add missing deps ([996bee2](https://github.com/hirosystems/eslint-config/commit/996bee23c11e48b8b288c2d932f51bedc7d31563))

janniks committed:
chore: update deps

janniks committed:
feat: switch to typescript

semantic-release-bot committed:
chore(release): 3.0.0-develop.3

## [3.0.0-develop.3](https://github.com/hirosystems/eslint-config/compare/v3.0.0-develop.2...v3.0.0-develop.3) (2025-06-16)

### Features

* switch to typescript ([8f98dbe](https://github.com/hirosystems/eslint-config/commit/8f98dbe2655b9e083c353e58baa4ecd06cf80028))

janniks committed:
fix: update module type
